test/e2e/network/service.go

```diff
@@ -2716,7 +2716,7 @@ var _ = SIGDescribe("Services", func() {
 		framework.ExpectNoError(verifyServeHostnameServiceDown(cs, host, svcHeadlessIP, servicePort))
 	})
 
-	ginkgo.It("should be rejected when no endpoints exist", func() {
+	ginkgo.It("should be rejected when no endpoints exist (kube-proxy)", func() {
```

Comment:
I have doubts on this one, this is testing behaviour, and I personally find it appropriate.

So,  if I create a service, the ClusterIP is assigned synchronously, but the endpoints are not, the are asynchronous, so my pods can start to send traffic to the ClusterIP and that traffic can be blackholed until the endpoints are ready.

This test checks that, if there are no endpoints, and I try to connect to the ClusterIP, the connection fails with an `ECONNREFUSED - Connection refused by server` error, is not checking how this is implemented.

I'll defer to @thockin , but I think that this behaviour should be implemented for Services.

Reply:
Ok, fair point, my thought on this was from the mailing list thread last week from @danwinship .... 

```
it's testing that kube-proxy  behaves as expected. It's true that a third party might happen to  interfere with this behavior, but that's not why the test is there. (It was added to the e2e suite after we discovered that a change to another  part of the iptables rule generation was causing the REJECT rules to be  bypassed.) 
``` 

maybe he can make the final call.

Reply:
I'm fine with one or another :+1:

Reply:
ok ill keep for now then

Reply:
It's not specifically a "kube-proxy" behavior; it's expected that all service proxy implementations will behave that way. eg, ovn-kubernetes (which doesn't use kube-proxy) used to fail this test, but then we fixed it.